publish/directory/openapi.go

```diff
@@ -1,11 +1,64 @@
 package openapi
 
-import "embed"
+import (
+	_ "embed"
+	"net/http"
+	"sync"
+	"text/template"
+
+	"github.com/rs/zerolog"
+)
+
+type Server struct {
+	Scheme string
+	Host   string
+}
 
 //go:embed openapi.json
-var staticAssets embed.FS
+var staticString string
+
+var buildTemplate = sync.OnceValues(func() (*template.Template, error) {
+	return template.New("openapi.json").Parse(staticString)
+})
+
+// Static string value of the openapi.json file.
+func Static() string {
+	return staticString
+}
+
+// Handler to serve the OpenAPI specification file.
+func OpenApiHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Add("Content-Type", "application/json")
+
+	template, err := buildTemplate()
+	if err != nil {
+		zerolog.Ctx(r.Context()).Err(err)
```

Comment:
I don't think this will actually write anything to the log. You need to either add message using `Msg()` or `Msg()` or if you don't want any message you can call `Send()`. But one of the three is required.